unblob/handlers/archive/zip.py

```diff
@@ -83,15 +112,53 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]
             file.seek(offset, io.SEEK_SET)
             end_of_central_directory = self.parse_header(file)
 
-            # the EOCD offset is equal to the offset of CD + size of CD
-            end_of_central_directory_offset = (
-                start_offset
-                + end_of_central_directory.offset_of_cd
-                + end_of_central_directory.central_directory_size
-            )
-
-            if offset == end_of_central_directory_offset:
-                break
+            if self.is_zip64_eocd(end_of_central_directory):
+                found_eocd = False
+
+                file.seek(start_offset, io.SEEK_SET)
+                for eocd_locator_offset in iterate_patterns(
+                    file, struct.pack("<I", ZIP64_EOCD_LOCATOR_HEADER)
+                ):
+                    file.seek(eocd_locator_offset, io.SEEK_SET)
+                    eocd_locator = (
+                        self.cparser_le.zip64_end_of_central_directory_locator_t(file)
+                    )
+                    logger.debug(
+                        "eocd_locator", eocd_locator=eocd_locator, _verbosity=3
+                    )
+
+                    # ZIP64 EOCD locator is right before the EOCD record
+                    if eocd_locator_offset + len(eocd_locator) == offset:
+                        file.seek(start_offset + eocd_locator.offset_of_cd)
+                        zip64_eocd = self.cparser_le.zip64_end_of_central_directory_t(
+                            file
+                        )
+                        logger.debug("zip64_eocd", zip64_eocd=zip64_eocd, _verbosity=3)
+
+                        if zip64_eocd.signature != ZIP64_EOCD_SIGNATURE:
+                            raise InvalidInputFormat(
+                                "Missing ZIP64 EOCD header record header in ZIP chunk."
+                            )
+                        end_of_central_directory = zip64_eocd
+                        found_eocd = True
+                        break
+
+                if found_eocd:
+                    break
```

Comment:
Just a small thing:

This whole zip64 finding and parsing could be in a new, separate function, making the core logic simpler, like:

```suggestion
            if self.is_zip64_eocd(end_of_central_directory):
                end_of_central_directory = self._parse_zip64(file, start_offset, offset)
                break
```

Reply:
Good point. Done.